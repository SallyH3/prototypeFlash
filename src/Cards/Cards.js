import React, { Component } from 'react';
import Card from '../Card/Card.js';
import Data from '../Data.js';

export default class Cards extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputBox: ''
    }
  }

  componentDidMount() {
    console.log(this.props)
  }

  checkAnswer = () => {
    if(this.state.inputBox === this.props.cards[this.props.randomNum].answer) {
      console.log(this.state.inputBox, this.props.cards[this.props.randomNum].answer, 'right')
    } else {
      console.log(this.state.inputBox, this.props.cards[this.props.randomNum].answer,'wrong')
    }
  }

  checkInput = (event) => {
    this.setState({inputBox: event.target.value})
  }

  render() {
    const cardInfo = this.props.cards[this.props.randomNum]
    if (this.props.cards.length === 0) {
      return <div></div>
    } else {
      return (
        <section className='right-box'>
          <section className='right-box-cards-container'>
            <article className='question'>
              {this.props.cards[this.props.randomNum].question}
            </article>
            <article className='right-controls'>
              <input onChange={this.checkInput} value={this.state.inputBox} className='card-input' type='text' placeholder='.reduce()'></input>
              <button onClick={this.checkAnswer} className='submit-answer'>Submit answer</button>
            </article>
            <footer className='footer-container'>
              <p className="learn-more">
                <a
                  href={cardInfo.link}
                  className='link'
                  target='_blank'
                  rel='noopener noreferrer'
                >
                  Learn more
                </a>
              </p>
            </footer>
          </section>
        </section>
      )
    }
  }
}