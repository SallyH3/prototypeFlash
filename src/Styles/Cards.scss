.right-box {
  background-color: $white;
  font-family: $primary;
  font-size: 1.3em;
  height: 100%;
  margin-top: 8em;
  padding: .8em;
  text-align: center;
  width: 50%;
}

.right-box-cards-container {
  border-radius: .5em;
  border: .1em solid $gray;
  height: 12em;
  margin-left: .4em;
  width: 27.5em;
}

.question {
  margin-top: 1.5em;
}

.right-controls {
  margin-top: 2.5em;
}

.card-input {
  border: 1px solid $gray;
  border-right: none;
  border-radius: .5em 0 0 .5em;
  font-size: .6em;
  height: 3em;
  padding-left: .7em;
  margin-top: 2px;
  vertical-align: bottom;
  width: 15em;
}

.submit-answer {
  border: 1px solid $gray;
  border-radius: 0 .5em .5em 0;
  border-left: none;
  background-color: $lavender;
  cursor: pointer;
  font-family: $primary;
  height: 3.4em;
  letter-spacing: .1em;
  width: 8em;
  &:hover {
    color: $white;
    background-color: hsla(263, 44%, 48%, 1);
    transition: all 0.3s ease-in-out;
  }
}

.footer-container {
  display: flex;
  justify-content: space-between;
  margin-top: -1em;
}

.type-text {
  font-size: .9em;
  margin-left: 1em;
  margin-top: 4em;
  text-align: left;
}

.learn-more {
  border: none;
  border-radius: .5em;
  color: $main-color;
  cursor: pointer;
  font-family: $primary;
  font-size: .7em;
  height: 3.4em;
  letter-spacing: .1em;
  margin-right: 2em;
  margin-top: 6em;
  width: 8em;
}