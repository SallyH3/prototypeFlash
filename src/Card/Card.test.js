import React from 'react';
import Cards from './Cards';
import { shallow } from 'enzyme';

const mockQuestion = [];

describe ('Cards', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Cards 
      question={mockQuestion}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it ('should have default states', () => {
    expect(wrapper.state()).toEqual({
      inputBox: '',
      question: [],
      correct: null
    })
  })
})