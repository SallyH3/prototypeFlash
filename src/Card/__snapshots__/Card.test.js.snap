// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<section
  className="right-box"
>
  <section
    className="right-box-cards-container"
  >
    <article
      className="question"
    />
    <article
      className="right-controls"
    >
      <input
        className="card-input"
        onChange={[Function]}
        placeholder="ex: .reduce()"
        type="text"
        value=""
      />
      <button
        className="submit-answer"
        onClick={[Function]}
      >
        Submit answer
      </button>
      <p
        className="answer-text"
      />
    </article>
    <footer
      className="footer-container"
    >
      <p
        className="learn-more"
      >
        <a
          className="link"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
        </a>
      </p>
      <p
        className="study-list-text"
      >
        Number wrong 
        0
      </p>
    </footer>
  </section>
</section>
`;
